src/ConstructiveSolidGeometry/SurfacePrimitives/Rectangle.jl

```diff
@@ -226,4 +226,32 @@ function get_vertices(r::RectangleZ{T}) where {T}
     CartesianPoint{T}(lMax, wMin, r.loc),
     CartesianPoint{T}(lMin, wMax, r.loc),
     CartesianPoint{T}(lMax, wMax, r.loc))
+end
+
+_get_rectangle_coordinates(point::CartesianPoint, r::RectangleX) = (point.y, point.z, point.x)
+_get_rectangle_coordinates(point::CartesianPoint, r::RectangleY) = (point.x, point.z, point.y)
+_get_rectangle_coordinates(point::CartesianPoint, r::RectangleZ) = (point.x, point.y, point.z)
+
+_get_rectangle_coordinates(point::CylindricalPoint, r::Rectangle) = _get_rectangle_coordinates(CartesianPoint(point), r)
+
+function distance_to_surface(point::AbstractCoordinatePoint{T}, r::Rectangle{<:Any, T})::T where {T}
+    l, w, d = _get_rectangle_coordinates(point, r)
+    lMin::T, lMax::T = get_l_limits(r)
+    wMin::T, wMax::T = get_w_limits(r)
+    if l in lMin..lMax && w in wMin..wMax
+        return abs(d - r.loc)
+    else 
+        if lMin ≤ l ≤ lMax
+            y = w < wMin ? wMin : wMax
+            return hypot(w - y, d - r.loc)
+        elseif wMin ≤ w ≤ wMax
+            x = l < lMin ? lMin : lMax
+            return hypot(l - x, d - r.loc)
+        else
+            lnear = l < lMin ? lMin : lMax
+            wnear = w < wMin ? wMin : wMax
+            return hypot(l - lnear, w - wnear, d - r.loc)
```

Comment:
How about `lnear = min(max(l, lMin), lMax)` and `wnear = min(max(w, wMin), wMax)` ?
This should cover all cases as `lnear = min(max(l, lMin), lMax)` is:
- `lMin` if `l < lMin`
- `l` if `lMin < l < lMax`
- `lMax` if `l > lMax`

The result in all cases would still be `hypot(l - lnear, w - wnear, d - r.loc)`?
as `l - lnear` will be `0` in the case `lMin < l < lMax`.

Lines 241 - 254 could then become:
```julia
return hypot(l - min(max(l, lMin), lMax), w - min(max(w, wMin), wMax), d - r.loc)
```

Not tested for performance.

Reply:
This for sure would work and its shorter. But it I explicitly wrote it the way it is to minimize number of operations